### ISSUE TITLE
garbage collect http agent pods when workflow is suspended by suspend templates

### ISSUE THREAD

**Issue by taehwoi** [argoproj/argo-workflows#13216]
# Summary

What change needs making?
Agent pod should be queued for clean up if the workflow has been suspended for x amount of time.

Currently, a workflow using http template will create an agent pod. The agent pod lives until the pipeline is finished.
If the pipeline has been suspended for some duration, delete the agent pod, and reschedule it when the workflow begins another http template.

This behavior may be configurable, so that users can choose between resource efficiency and responsiveness.
## Use Cases

When would you use this?
using http template with suspend template in a workflow. 

---

**Message from the maintainers**:

Love this feature request? Give it a 👍. We prioritise the proposals with the most 👍.

**Comment by agilgur5** [comment]
> This behavior may be configurable, so that users can choose between resource efficiency and responsiveness.

I don't think we'd make it configurable as that would entail more features to support and maintain.

Otherwise, this feature makes sense to me; I think agent Pods and suspended Workflows were just never accounted for

**Comment by taehwoi** [comment]
@agilgur5
I agree, I don't think waiting for the agent pod to be live again would be a concern when the pipeline has been already suspended for some time.

**Comment by jswxstw** [comment]
> If the workflow has been suspended for some duration, delete the agent pod

The waiting condition is not very reliable, the agent pod can only be deleted after all running tasks are completed when workflow is suspending.
